src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs

```diff
@@ -364,15 +415,18 @@ public override void ExecuteCmdlet()
                         WriteObject(encryptionStatus);
                         break;
                     case OSType.Linux:
-                        VirtualMachine virtualMachineResponse = this.ComputeClient.ComputeManagementClient.VirtualMachines.GetWithInstanceView(
-                            this.ResourceGroupName, VMName).Body;
-                        VirtualMachineExtension parameters = GetVmExtensionParameters(virtualMachineResponse, osType);
-
-                        this.VirtualMachineExtensionClient.CreateOrUpdateWithHttpMessagesAsync(
-                            this.ResourceGroupName,
-                            this.VMName,
-                            this.Name,
-                            parameters).GetAwaiter().GetResult();
+                        if (!IsExtensionInstalled(osType))
```

Comment:
@cormacpayne On Windows we do not perform a query operation anyway so we do not need to check if extension is installed.

Inside the function itself, I added logic for Windows just to stay consistent in case the function is needed in the future.